In [1]:
import pandas as pd
import matplotlib.pyplot as plt
import numpy as np

In [2]:
ds = pd.read_csv('MLR_numeric.csv')

In [3]:
ds.head()

S&P department (spend)  Admin department(spend)  \
0                  101184                    70674   
1                   70132                    52030   
2                  149539                    98535   
3                   75821                    53649   
4                   97730                    68667   

   Operation department(spend)  Profit  
0                        97009  259480  
1                        63587  137019  
2                       199994  495552  
3                        66500  165057  
4                        91754  247841

S&P department (spend)  Admin department(spend)  \
0                  101184                    70674   
1                   70132                    52030   
2                  149539                    98535   
3                   75821                    53649   
4                   97730                    68667   

   Operation department(spend)  Profit  
0                        97009  259480  
1                        63587  137019  
2                       199994  495552  
3                        66500  165057  
4                        91754  247841

In [4]:
ds.shape

(50, 4)

(50, 4)

In [5]:
x = ds.iloc[:, :-1].values
y = ds.iloc[:, 3].values

In [6]:
corr = ds.corr()
print(corr)

                             S&P department (spend)  Admin department(spend)  \
S&P department (spend)                     1.000000                 0.984560   
Admin department(spend)                    0.984560                 1.000000   
Operation department(spend)                0.976437                 0.980723   
Profit                                     0.982576                 0.993823   

                             Operation department(spend)    Profit  
S&P department (spend)                          0.976437  0.982576  
Admin department(spend)                         0.980723  0.993823  
Operation department(spend)                     1.000000  0.975566  
Profit                                          0.975566  1.000000  
                             S&P department (spend)  Admin department(spend)  \
S&P department (spend)                     1.000000                 0.984560   
Admin department(spend)                    0.984560                 1.000000   
Operation depa

## Use Multiple Linear Regression

In [7]:
from sklearn.linear_model import LinearRegression

regressor = LinearRegression()
regressor.fit(x,y)

LinearRegression()

LinearRegression()

In [8]:
print('The coefficient is:', regressor.coef_)
print('The y-intercept is:', regressor.intercept_)

The coefficient is: [ 0.62343078  6.49776395 -0.04246029]
The y-intercept is: -235096.75443959906
The coefficient is: [ 0.62343078  6.49776395 -0.04246029]
The y-intercept is: -235096.75443959906


The multiple linear equation is
$ \text{Profit} = -235096.75 +0.6234 \text{ S&P department (spend)} \\+ 6.4977 \times \text{Admin department(spend)} -0.04246 \times \text{Operation department(spend)} $

---

## Predict

Question


What is the profit for 2024 if the total spending amount from S&P, Admin and Operation departments 
are RM90020, RM55200 and RM94230 respectively?

In [9]:
a = [[90020, 55200, 94230]]
y_predict = regressor.predict(a)
print(y_predict)

[175700.02117444]
[175700.02117444]


In [10]:
a = [[100000, 40000, 120000]]
y_predict = regressor.predict(a)
print(y_predict)

[82061.64673662]
[82061.64673662]


In [11]:
a = [[100000, 50000, 120000]]
y_predict = regressor.predict(a)
print(y_predict)

[147039.28619657]
[147039.28619657]


In [12]:
a = [[100000, 60000, 120000]]
y_predict = regressor.predict(a)
print(y_predict)

[212016.92565652]
[212016.92565652]
